### PR TITLE
Limit frame-ancestors to requesting URI for embedded shops

### PR DESCRIPTION
**Pull request by lin-d-hop**
#### What? Why?

#1953 
Proposing an alternative to listing every allowed parent in every header. When debugging iframe issues it felt inappropriate to be able to see the list of all URIs that can embed OFN shopfronts from a clients website.

#### What should we test?

Not ready for testing. WIP

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -57,10 +57,10 @@ def restrict_iframes
   def enable_embedded_shopfront
     whitelist = Spree::Config[:embedded_shopfronts_whitelist]
     return unless Spree::Config[:enable_embedded_shopfronts] && whitelist.present?
-    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test? && !Rails.env.development?
+    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test? && !Rails.env.development? && !whitelist.include?(URI(request.referer).host)
```
Comment (sauloperez) [inline app/controllers/application_controller.rb]: Did we consider extracting this logic from the application controller? This line is getting hairy over time (the last two commits touch it) and thus becoming harder to change. Besides, the embedded stuff adds an extra responsibility the controller. We should extract it from here as per the single responsibility principle.

**Comment by daniellemoorhead** [comment]
@lin-d-hop Maikel is on holiday until early December. Did you want to wait for him to get back to review this or plan B is Rob looking at it. Your choice 🙂

**Comment by lin-d-hop** [comment]
@daniellemoorhead not in a hurry. I'm happy to follow up with @matt-yorkley when he's back

**Comment by lin-d-hop** [comment]
Replaced by #2026

**Pull request closed without merging**